Cabal/src/Distribution/Simple/Setup.hs

```diff
@@ -1961,7 +1974,8 @@ instance Semigroup BuildFlags where
 
 data ReplOptions = ReplOptions {
     replOptionsFlags :: [String],
```

Comment:
This is in `Cabal`, so these options will only be passed to one component. But I suppose the question is what happens if you pass `--repl-options` to cabal-install when starting a multi session.

In this case if you pass `--repl-options` it will pass the options to each component individually (in the response file) rather than supplying the arguments to ghc directly at the top-level invocation. Perhaps that is surprising behaviour so I will open a ticket where people can discuss what they think should happen.